var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -236,104 +231,60 @@ def libs(self):
+                                        spec['libaec'].prefix.include))
+            args.append(self.define('SZIP_DIR', spec['libaec'].prefix.lib)) 
+
+        args.append(self.define_from_variant('HDF5_ENABLE_PARALLEL', 'mpi'))
+
+        args.append(self.define_from_variant('HDF5_ENABLE_THREADSAFE', 'threadsafe'))
+
+        args.append(self.define_from_variant('HDF5_BUILD_HL_LIB', 'hl'))
+
+        args.append(self.define_from_variant('HDF5_BUILD_CPP_LIB', 'cxx'))
 
-        if '+shared' in self.spec:
-            extra_args.append('--enable-shared')
+        args.append(self.define_from_variant('HDF5_BUILD_FORTRAN', 'fortran'))
+
+        args.append(self.define_from_variant('HDF5_BUILD_JAVA', 'java'))
+
+        args.append(self.define_from_variant('HDF5_BUILD_TOOLS', 'tools'))
+
+        if self.run_tests:
+            args.append('-DBUILD_TESTING=ON')
         else:
-            extra_args.append('--disable-shared')
-            extra_args.append('--enable-static-exec')
-
-        if '+pic' in self.spec:
-            extra_args.extend([
-                'CFLAGS='   + self.compiler.cc_pic_flag,
-                'CXXFLAGS=' + self.compiler.cxx_pic_flag,
-                'FCFLAGS='  + self.compiler.fc_pic_flag,
-            ])
-
-        # Fujitsu Compiler dose not add  Fortran runtime in rpath.
-        if '+fortran %fj' in self.spec:
-            extra_args.append('LDFLAGS=-lfj90i -lfj90f -lfjsrcinfo -lelf')
-
-        if '+mpi' in self.spec:
-            # The HDF5 configure script warns if cxx and mpi are enabled
-            # together. There doesn't seem to be a real reason for this, except
-            # that parts of the MPI interface are not accessible via the C++
-            # interface. Since they are still accessible via the C interface,
-            # this is not actually a problem.
-            extra_args += ['--enable-parallel',
-                           'CC=%s' % self.spec['mpi'].mpicc]
-
-            if '+cxx' in self.spec:
-                extra_args.append('CXX=%s' % self.spec['mpi'].mpicxx)
-
-            if '+fortran' in self.spec:
-                extra_args.append('FC=%s' % self.spec['mpi'].mpifc)
-
-        return extra_args
-
-    @run_after('configure')
-    def patch_postdeps(self):
-        if '@:1.8.14' in self.spec:
-            # On Ubuntu14, HDF5 1.8.12 (and maybe other versions)
-            # mysteriously end up with "-l -l" in the postdeps in the
-            # libtool script.  Patch this by removing the spurious -l's.
-            filter_file(
-                r'postdeps="([^"]*)"',
-                lambda m: 'postdeps="%s"' % ' '.join(
-                    arg for arg in m.group(1).split(' ') if arg != '-l'),
-                'libtool')
+            args.append('-DBUILD_TESTING=OFF')
+
+        if spec.variants['api'].value != 'default':
+            args.append(
+                '-DDEFAULT_API_VERSION={0}'.format(
+                    spec.variants['api'].value))
```

Comment:
A little tweak to suggest if I may:
```suggestion
        api = spec.variants['api'].value
        if api != 'default':
            args.append(self.define('DEFAULT_API_VERSION', api))
```